apps/omg_watcher_rpc/lib/web/controllers/block.ex

```diff
@@ -43,4 +44,22 @@ defmodule OMG.WatcherRPC.Web.Controller.Block do
       |> api_response(conn, :blocks)
     end
   end
+
+  @doc """
+  Executes stateful and stateless validation of a block.
+  """
+  def validate_block(conn, params) do
+    with {:ok, block} <- Validator.BlockValidator.parse_to_validate(params),
+         {:ok, _block} <- stateless_validate(block) do
+      api_response(block, conn, :validate_block)
+    end
+  end
+
+  @spec stateless_validate(Block.t()) :: any
+  defp stateless_validate(submitted_block) do
+    with {:ok, recovered_transactions} <- Validator.BlockValidator.verify_transactions(submitted_block.transactions),
+         {:ok, block} <- Validator.BlockValidator.verify_merkle_root(submitted_block, recovered_transactions) do
+      {:ok, block}
+    end
+  end
```

Comment:
I think this should be moved to `OMG.Watcher` along with my other comment in https://github.com/omgnetwork/elixir-omg/pull/1668/files#diff-fbb5622ef4f2eb45b525f9d2cb8014dc

And remaining here would be something like:

```
  def validate_block(conn, params) do
    with {:ok, block} <- Validator.BlockValidator.parse_to_validate(params),
         {:ok, _block} <- OMG.Watcher.Block.stateless_validate(block) do
      api_response(block, conn, :validate_block)
    end
  end
```